streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -644,8 +644,10 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester
         val mappedStream = fileStream.map(s => {
           val i = s.toInt
           if (i == 3) {
-            while (CheckpointSuite.batchThreeShouldBlockIndefinitely) {
-              Thread.sleep(Long.MaxValue)
+            if (CheckpointSuite.batchThreeShouldBlockIndefinitely) {
+              // It's not a good idea to let the thread run forever
+              // as resource won't be correctly released
+              Thread.sleep(6000)
```

Comment:
OK, how about changing the text to "should block a long time" then? as long as this is more than long enough for the driver to shutdown it seems to be in line with the intent of the test.